pkg/backend/httpstate/backend.go

```diff
@@ -384,15 +398,23 @@ func (m defaultLoginManager) Current(
 	existingAccount, err := workspace.GetAccount(cloudURL)
 	if err == nil && existingAccount.AccessToken != "" &&
 		(accessToken == "" || existingAccount.AccessToken == accessToken) {
-		// If the account was last verified less than an hour ago, assume the token is valid.
 		valid := true
+		existingAccount, err = refreshAuthentication(cloudURL, insecure, existingAccount)
+		if err != nil {
+			return nil, err
+		}
 		username := existingAccount.Username
 		organizations := existingAccount.Organizations
 		tokenInfo := existingAccount.TokenInformation
+		authContext := existingAccount.AuthContext
+		// If the account was last verified less than an hour ago, assume the token is valid.
```

Comment:
Off-topic (because the comment was already there): this comment has had me thinking and given me more questions than answers.

- This branch doesn't really assume that the token is valid, no? Won't `getAccountDetails` check again? What do we really mean here? My assumption: "because the token was verified less than an hour ago, we don't try to check for its validity again"
    - I would rather have the comment explain the "why": is verifying the token again expensive/slow?
    - What if the user has refreshed the credentials in some other way that doesn't affect the return of `LastValidatedAt`?

Suggestion: remove the comment or replace it with one that says why it's OK to not verify the token again.